optimum/onnxruntime/configuration.py

```diff
@@ -604,73 +608,222 @@ def tensorrt(
             The default value is set to `False` since this approximation might slightly impact the model's accuracy.
         use_mask_index (`bool`, defaults to `False`):
             Whether to use mask index instead of raw attention mask in the attention operator.
         no_attention_mask (`bool`, defaults to `False`):
             Whether to not use attention masks. Only works for bert model type.
-        disable_embed_layer_norm (`bool`, defaults to `True`):
-            Whether to disable EmbedLayerNormalization fusion.
-            The default value is set to `True` since this fusion is incompatible with ONNX Runtime quantization.
         disable_shape_inference (`bool`, defaults to `False`):
             Whether to disable symbolic shape inference.
             The default value is set to `False` but symbolic shape inference might cause issues sometimes.
     """
 
-    optimization_level: int = 1
+    optimization_level: Optional[int] = None
+    general_optimization_level: int = 1
+
     optimize_for_gpu: bool = False
+
     fp16: bool = False
-    optimize_with_onnxruntime_only: bool = False
-    disable_gelu: bool = False
-    disable_layer_norm: bool = False
-    disable_attention: bool = False
-    disable_skip_layer_norm: bool = False
-    disable_bias_skip_layer_norm: bool = False
-    disable_bias_gelu: bool = False
+
+    optimize_with_onnxruntime_only: Optional[bool] = None
+    enable_transformers_specific_optimizations: bool = True
+
+    disable_gelu: Optional[bool] = None
+    disable_gelu_fusion: bool = False
+
+    disable_layer_norm: Optional[bool] = None
+    disable_layer_norm_fusion: bool = False
+
+    disable_attention: Optional[bool] = None
+    disable_attention_fusion: bool = False
+
+    disable_skip_layer_norm: Optional[bool] = None
+    disable_skip_layer_norm_fusion: bool = False
+
+    disable_bias_skip_layer_norm: Optional[bool] = None
+    disable_bias_skip_layer_norm_fusion: bool = False
+
+    disable_bias_gelu: Optional[bool] = None
+    disable_bias_gelu_fusion: bool = False
+
+    disable_embed_layer_norm: Optional[bool] = None
+    disable_embed_layer_norm_fusion: bool = True
+
     enable_gelu_approximation: bool = False
     use_mask_index: bool = False
     no_attention_mask: bool = False
-    disable_embed_layer_norm: bool = True
     disable_shape_inference: bool = False
 
+    def __post_init__(self):
+        def deprecate_renamed_attribute(old_name, new_name, mapping_func=None):
+            if getattr(self, old_name, None) is not None:
+                warnings.warn(
+                    f"{old_name} will be deprecated soon, use {new_name} instead.",
+                    FutureWarning,
+                )
+                if mapping_func is None:
+
+                    def identity(x):
+                        return x
+
+                    mapping_func = identity
+                setattr(self, new_name, mapping_func(getattr(self, old_name)))
```

Comment:
Maybe we could put the deprecation warning after setting the value of the new attribute and add to the warning message something like: `f"{new_name} is set to {mapping_func(getattr(self, old_name))}"`.